normandy/recipes/models.py

```diff
@@ -739,6 +740,23 @@ def default():
             if arguments["rolloutSlug"] not in rollout_slugs:
                 errors["slug"] = self.errors["rollout_slug_not_found"]
 
+        elif self.name == "show-heartbeat":
+            # Survey ID should be unique across all recipes
+            other_recipes = Recipe.objects.filter(latest_revision__action=self)
+            if revision.recipe and revision.recipe.id:
+                other_recipes = other_recipes.exclude(id=revision.recipe.id)
+            # Note, `recipe.arguments` refers to `recipe.current_revision.arguments`
```

Comment:
I think the placement of this comment is a bit misleading. This note applies to most of the checks in this function. However putting the comment here makes it seem like it applies uniquely to `show-heartbeat`. Can we move the comment to a more inclusive place?

Overall, I like to see us move away from the helpers defined on Recipe, and instead access revisions directly. This matches well with the sort of changes we made in the v3 API, and I think it will help us think more clearly about things. I'm not sure if it would help here though.

Reply:
> Can we move the comment to a more inclusive place?

Done. 

> ...move away from the helpers defined on Recipe, and instead access revisions directly. 

Yes please! I think I did it the way I did it because of the historical context (the lines above). 

I could easily change it to be:

```python
other_revisions = RecipeRevision.objects.filter(action=self).exclude(
    recipe_id=revision.recipe.id
)
for other_revision in other_revisions:
    if other_revision.arguments["surveyId"] == arguments["surveyId"]:
        errors["surveyId"] = self.errors["duplicate_survey_id"]
        break
```
Roughly. An option is to not do `errors["surveyId"] = self.errors["duplicate_survey_id"] + f' (used in {other_revision.recipe.name})'. 

One caveat is that a surveyId, that I want to use, is lingering inside a revision (in a different recipe) that is long ago replaced with a better revision. Does that make sense?